lib/modules/platform/bitbucket/index.ts

```diff
@@ -949,3 +954,81 @@ export async function mergePr({
   }
   return true;
 }
+
+export async function getReleaseNotesMd(
+  repository: string,
+  sourceDirectory?: string
+): Promise<ChangeLogFile | null> {
+  logger.trace('bitbucket.getReleaseNotesMd()');
+
+  const repositorySourceURl = joinUrlParts(
+    '/2.0/repositories',
+    repository,
+    'src'
+  );
+
+  const rootFiles =
+    await bitbucketHttp.getUnpaginatedJson<BitbucketSourceResults>(
+      repositorySourceURl
+    );
+
+  const allFiles = rootFiles.filter((f) => f.type === 'commit_file');
+
+  let files: BitbucketSourceResults[] = [];
+
+  if (!files.length) {
+    files = allFiles.filter((f) => changelogFilenameRegex.test(f.path));
+  }
+  if (!files.length) {
+    logger.trace('no changelog file found');
+    return null;
+  }
+
+  const changelogFile = files.shift()!;
+
+  if (files.length !== 0) {
+    logger.debug(
+      `Multiple candidates for changelog file, using ${changelogFile.path}`
+    );
+  }
+
+  const fileRes = await bitbucketHttp.get(
+    joinUrlParts(
+      repositorySourceURl,
+      changelogFile.commit.hash,
+      changelogFile.path
+    )
+  );
+
+  const changelogMd = fileRes.body + '\n#\n##';
+  return { changelogFile: changelogFile.path, changelogMd };
+}
+
+export async function getTags(repository: string): Promise<string[]> {
```

Comment:
this should be a bitbucket tags datasource

Reply:
updated